### DISCUSSION TITLE
Flash Firmware Problem

### DISCUSSION THREAD

**Discussion by aislanmds** [doudar/SmartSpin2k#604] [General]
Hello everyone, I have an ESP32 Wroon 32 board, and I'm having trouble flashing the firmware. I can write code to the board from Arduino IDE, for example blinking the D2 LED. But I can't flash the firmware from SmartSpin2kFlasher.exe. I get the error :
![error](https://github.com/user-attachments/assets/143661a8-6c05-49af-b897-969c8cd83a67)
Could someone help me?

Thanks

**Comment by eMadman** [comment]
On those ESP32 devkits.  The devkits don't always have the capacitor you need for it to automatically go into bootloader mode. you might need to press and hold the boot pin, then tap and release the en pin.  That'll restart it and get it into  bootloader mode.

Also be sure to install the CP210x drivers from [here](https://www.silabs.com/developer-tools/usb-to-uart-bridge-vcp-drivers?tab=downloads)

**Comment by aislanmds** [comment]
Thanks for the help. I tried with your information, holding the boot button, and a light touch on the EN button. I was not successful.

**Reply by eMadman** [reply]
Try the [FTDI  Virtual COM port driver as well](https://ftdichip.com/drivers/vcp-drivers/)... and also double check the CP210x driver.  There's two for windows, try the other one.  

Are you on a V2 with an esp32 devkit or a v3 PCB kit?

**Comment by aislanmds** [comment]
I'm using these two boards, with CH340 and CP2102. Just remember that I can flash new projects using the Arduino IDE, but I can't send the .bin files using the program provided. I'll try the VCP driver and the other C2102 drive option for Windows and then I'll make a new comment. Thanks.
![Screenshot_20241212_054326_AliExpress](https://github.com/user-attachments/assets/080f114b-7d68-49f0-a030-74903aee922f)
https://a.aliexpress.com/_mNKzkAj

**Reply by eMadman** [reply]
I'm familiar with these devkits from Aliexpress.  Before we switched to V3, I was getting a failure rate of over 30%.  Flashing not working... I would suggest also trying some different cables.  I've had several cheap Micro USB cables that were power only.

**Comment by aislanmds** [comment]
Thanks for the responses. The cables I use are of good quality too. When you had failures with these kits, were you still able to upload other projects via the Arduino IDE?

**Comment by aislanmds** [comment]
I was able to upload the .bin file using the ESP32 Download Tool software. Could you please let me know if I need to flash the 4 .bin files? 
I saw information that it is possible to flash multiple files at the same time. But when I try to do this, an error occurs. 
I also don't know what hexadecimal values I should enter for each .bin file to flash it using this software. Have you used this tool?

**Reply by doudar** [reply]
When the firmware first boots, it should partition the rest of your drive correctly. When you connect it to WiFi, it will download all of the other files.

**Comment by aislanmds** [comment]
Hello, how are you? Sorry for the delay in responding with my feedback.
I was successful in uploading the firmware using the esp flash download tool program, and these were the results. I used these settings from the screens below. In the end, I restarted the ESP, but I couldn't connect via bluetooth or wifi. I couldn't get feedback from the board to know if everything is ok. I appreciate any help
![Captura de tela 2024-12-14 154517](https://github.com/user-attachments/assets/8bbcea73-74b5-4474-b9d4-27902df392a4)

![Captura de tela 2024-12-14 153610](https://github.com/user-attachments/assets/45a0d2f6-19dc-4611-841f-586cc1db551b)

**Reply by aislanmds** [reply]
Arduino IDE serial monitor:
![Captura de tela 2024-12-14 185128](https://github.com/user-attachments/assets/99daa4b8-0d7f-4ef1-93a3-5db50493c81d)

**Comment by aislanmds** [comment]
Sorry, I closed the discussion unintentionally. I still haven't solved the problem. :(

**Comment by aislanmds** [comment]
Problem solved, I was able to upload it by changing the hexadecimals related to the memories.
